umh-core/pkg/fsm/dataflowcomponent/fsm_callbacks.go

```diff
@@ -26,40 +27,38 @@ func (instance *DataflowComponentInstance) registerCallbacks() {
 	// Basic operational state callbacks
 	instance.baseFSMInstance.AddCallback("enter_"+OperationalStateStarting, func(ctx context.Context, e *fsm.Event) {
 		instance.baseFSMInstance.GetLogger().Infof("Entering starting state for %s", instance.baseFSMInstance.GetID())
+		instance.stateEntryTime[OperationalStateStarting] = time.Now()
```

Comment:
can we instead of time, work with ticks instead? time.now makes it hard to mock and test

also what about the discussion we had here https://github.com/united-manufacturing-hub/united-manufacturing-hub/pull/1928#discussion_r2042211999

that would likely be the more scalable solution